jetstream.go

```diff
@@ -0,0 +1,332 @@
+package nats
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	AckAck      = []byte("+ACK")
+	AckNak      = []byte("-NAK")
+	AckProgress = []byte("+WPI")
+	AckNext     = []byte("+NXT")
+	AckTerm     = []byte("+TERM")
+)
+
+// JetStreamMsgMetaData is metadata related to a JetStream originated message
+type JetStreamMsgMetaData struct {
+	stream    string
+	consumer  string
+	parsed    bool
+	delivered int
+	sSeq      int
+	cSeq      int
+	ts        time.Time
+}
+
+func (m *JetStreamMsgMetaData) Stream() string {
+	return m.stream
+}
+
+func (m *JetStreamMsgMetaData) Consumer() string {
+	return m.consumer
+}
+
+func (m *JetStreamMsgMetaData) Valid() bool {
+	return m.parsed
+}
+
+func (m *JetStreamMsgMetaData) Delivered() int {
+	return m.delivered
+}
+
+func (m *JetStreamMsgMetaData) StreamSequence() int {
+	return m.sSeq
+}
+
+func (m *JetStreamMsgMetaData) ConsumerSequence() int {
+	return m.cSeq
+}
+
+func (m *JetStreamMsgMetaData) PublishTime() time.Time {
```

Comment:
Still think we remove these and just make the meta struct a data struct and make them public.